api/envoy/admin/v3/config_dump.proto

```diff
@@ -184,6 +187,9 @@ message ClustersConfigDump {
 
     // The timestamp when the Cluster was last updated.
     google.protobuf.Timestamp last_updated = 3;
+    
+    // Set if the last update failed, cleared after the next successful update.
+    UpdateFailureState error_state = 4;
```

Comment:
Just to check my understanding here, this is how I think this will work:

If we are currently using version 1 of a resource because we NACKed version 2, then `DynamicCluster.version_info` will be "1", `DynamicCluster.cluster` will be the contents of version 1, and `DynamicCluster.error_state.version_info` will be "2".

Is that right?  We should probably add comments to all of these fields clarifying exactly what they will contain.

Note that the semantics of the individual fields may be different when the CSDS output comes from an xDS client rather than an xDS server, and these protos are used in both contexts, so the comments should make that clear.  The summary I gave above is for an xDS client only.

Reply:
Comment updated.

Yes, the example you described is exactly how this field gonna work. Also, I think the same mechanism works for xDS servers as well. CSDS is used by xDS servers to answer a specific client's configuration status, so it's valid to have the config dumped resource version match the in-effective version and have the rejected details stored separately. WDYT?

CC @fuqianggao

Reply:
We should also add a comment to the `UpdateFailureState.failed_configuration` field that it may not be populated on clients.  If we've NACKed the most recent version, we can record that version, but I don't think we want to record the entire failed resource.

Reply:
Updated the comment, basically saying this field is optional.